plugins/modules/katello_upload.py

```diff
@@ -151,27 +151,24 @@ def main():
     entity_dict['repository'] = module.find_resource_by_name('repositories', entity_dict['repository'], params=product_scope)
     repository_scope = {'repository_id': entity_dict['repository']['id']}
 
+    b_src = to_bytes(entity_dict['src'])
     filename = os.path.basename(entity_dict['src'])
 
-    checksum = hashlib.sha256()
-    with open(entity_dict['src'], 'rb') as contentfile:
-        for chunk in iter(lambda: contentfile.read(CONTENT_CHUNK_SIZE), b""):
-            checksum.update(chunk)
-    checksum = checksum.hexdigest()
+    checksum = module.sha256(entity_dict['src'])
```

Comment:
"oh!"

Reply:
also interesting how you have to pass the original `src` here, as `sha256 → digest_from_file` does `b_filename = to_bytes(filename, errors='surrogate_or_strict')` itself. but then again, shouldn't `to_bytes(to_bytes())` work?

Reply:
> `to_bytes(to_bytes())`

Yes i think that should do, but i think i have seen that failing. Maybe due to the missing `errors='surrogate_or_strict'`. And if they convert it anyway, why should i.

Reply:
yeah, it's just weird to read.